internal/handlers/pg/pgdb/query_documents.go

```diff
@@ -124,6 +106,82 @@ func (pgPool *Pool) QueryDocuments(ctx context.Context, querier pgxtype.Querier,
 	return fetchedChan, nil
 }
 
+// Explain returns document list with explain analyze results.
+// We don't expect much documents in results here.
+func (pgPool *Pool) Explain(ctx context.Context, querier pgxtype.Querier, sp SQLParam) ([]*types.Document, error) {
+	sql, err := pgPool.buildQuery(ctx, querier, sp)
+	if err != nil {
+		return nil, lazyerrors.Error(err)
+	}
+	rows, err := querier.Query(ctx, sql)
+	if err != nil {
+		return nil, lazyerrors.Error(err)
+	}
+	var res []*types.Document
+	for ctx.Err() == nil {
+		if !rows.Next() {
+			break
+		}
```

Comment:
I think we can put `rows.Next()` inside `for` condition as they both will interrupt `for` execution.
```suggestion
	for ctx.Err() == nil && rows.Next() {
```

Reply:
changed